stl/inc/chrono

```diff
@@ -3454,84 +3454,72 @@ namespace chrono {
             _All, // utc_clock
-            _File_time // _Negative before 1 January 2017, _All afterwards.
+            _File_time // _None before 2018-06-01, _All afterwards.
         };
 
         template <class _DurationType>
-        _NODISCARD bool _Make_time_point(_DurationType& _Dur, _LeapSecondRep _Leap) {
+        _NODISCARD bool _Make_time_point(_DurationType& _Dur, _Leap_second_rep _Leap) {
 
             const bool _Consistent{_Calculate_hour24() && _Calculate_year_fields() && _Calculate_ymd()};
             if (!_Consistent || !_Apply_duration_fields<_Parse_tp_or_duration::_Time_point>(_Dur)) {
                 return false;
             }
 
-            const _LeapSecondRep _Original_leap{_Leap};
-            if (_Leap == _LeapSecondRep::_File_time) {
-                const int _Year{*_Century + *_Two_dig_year};
-                _Leap = _Year <= 2016 ? _LeapSecondRep::_Negative : _LeapSecondRep::_All;
+            const _Leap_second_rep _Original_leap{_Leap};
+            if (_Leap == _Leap_second_rep::_File_time) {
+                _Leap =
+                    _Dur < file_clock::_Cutoff.time_since_epoch() ? _Leap_second_rep::_None : _Leap_second_rep::_All;
             }
 
-            if (_Second > (_Leap == _LeapSecondRep::_All ? 60 : 59)) {
+            if (_Second > (_Leap == _Leap_second_rep::_All ? 60 : 59)) {
                 return false;
             }
 
-            // A distinction has to be made here between clocks that tick monotonically, even around a positive leap
-            // second (everything except system_clock) and ones that have a special representation for leap seconds
-            // (utc_clock and, for negative leap seconds, system_clock). gps_clock and tai_clock, in particular, always
-            // monotonically count 86,400 seconds/day. The correct tick count, therefore, can be determined without
+            // A distinction has to be made here between clocks that always count 86,400 seconds/day (system_clock,
+            // tai_clock, and gps_clock) and ones that count leap seconds (utc_clock and, since 2018-06-01, file_clock).
+            // The correct tick count for the clocks that always count 86,400 seconds/day can be determined without
             // knowing whether any leap seconds have occurred, and there aren't any invalid times due to negative leap
             // second deletions.
-            //
-            // system_clock also has 86,400 seconds/day, but observes negative leap seconds by skipping them in its tick
-            // count. So leap second data is needed to check for a valid time, but not to calculate the tick count.
 
-            if (_Leap != _LeapSecondRep::_None) {
-                if (_Hour_24 == 23 && _Minute == 59 && _Second >= 59) {
+            if (_Leap == _Leap_second_rep::_All) {
+                constexpr bool _Can_rep_sec = _Can_represent<_DurationType, seconds>();
+                if constexpr (_Can_rep_sec) {

```

Comment:
I know this is existing, but can we turn this into an early return? Having the `else` case so far down makes it kind of hard to read:

```cxx
                if constexpr (!_Can_rep_sec) {
                    // Error if _Dur can't represent the above adjustment.
                    return false;
                }
```

also, there's no real reason here to make `_Can_rep_sec` an automatic variable, you could just

```cxx
                if constexpr (!_Can_represent<_DurationType, seconds>())
```